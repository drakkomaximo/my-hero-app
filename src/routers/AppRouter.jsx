import React, { useContext } from 'react'
import {
    BrowserRouter as Router,
    Switch
  } from "react-router-dom";
import { AuthContext } from '../auth/AuthContex';
import LoginScreen from '../components/login/LoginScreen';
import DashboardRoutes from './DashboardRoutes';
import PrivateRoutes from './PrivateRoutes';
import PublicRoute from './PublicRoute';

const AppRouter = () => {

  const { heroUser} = useContext( AuthContext )

    return (
        <Router>
          <div>
            <Switch>
              <PublicRoute 
                exact 
                path="/login" 
                component={ LoginScreen } 
                isAuthenticated={ heroUser.logged }
                />
              <PrivateRoutes 
                path="/" 
                component={ DashboardRoutes } 
                isAuthenticated={ heroUser.logged }
                />
            </Switch>
          </div>
        </Router>
      );
}

export default AppRouter
