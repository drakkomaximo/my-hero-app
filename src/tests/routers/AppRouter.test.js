import { mount } from "enzyme"
import { AuthContext } from "../../auth/AuthContex"
import AppRouter from "../../routers/AppRouter"

describe('Tests on <AppRouter />', () => {

    test('It must login if user is not autenticated', () => {

        const contextValue = {
            dispatch: jest.fn(),
            heroUser: {
                logged: false
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        )

        expect( wrapper ).toMatchSnapshot()
    })

    test('It must marvel component if user is autenticated', () => {
        const contextValue = {
            dispatch: jest.fn(),
            heroUser: {
                logged: true,
                name:'Andr√©s'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        )

        expect( wrapper.find('.navbar').exists() ).toBe(true)
    })
    
    
})
