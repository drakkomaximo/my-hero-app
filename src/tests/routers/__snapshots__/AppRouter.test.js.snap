// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter /> It must login if user is not autenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login Screen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;

exports[`Tests on <AppRouter /> It must login if user is not autenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login Screen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
